components/cloud/azure/src/azblob.rs

```diff
@@ -498,28 +494,44 @@ impl AzureStorage {
     }
 
     pub fn new(config: Config) -> io::Result<AzureStorage> {
+        let bucket = (*config.bucket.bucket).to_owned();
         // priority: explicit shared key > env Azure AD > env shared key
         if let Some(connection_string) = config.parse_plaintext_account_url() {
-            let bucket = (*config.bucket.bucket).to_owned();
-            let http_client = new_http_client();
-            let container_client = StorageAccountClient::new_connection_string(
-                http_client.clone(),
-                connection_string.as_str(),
-            )
-            .map_err(|e| io::Error::new(io::ErrorKind::InvalidInput, format!("{}", &e)))?
-            .as_storage_client()
-            .as_container_client(bucket);
+            #[cfg(test)]
+            #[cfg(feature = "azurite")]
+            {
+                debug_assert!(!connection_string.is_empty());
+                return Ok(AzureStorage {
+                    config,
+                    client_builder: Arc::new(SharedKeyContainerBuilder {
+                        container_client: Arc::new(
+                            ClientBuilder::emulator()
+                                .blob_service_client()
+                                .container_client(bucket)
+                                .clone(),
+                        ),
+                    }),
+                });
+            }
+            let account_name = config.get_account_name()?;
+            let storage_credentials = ConnectionString::new(&connection_string)
+                .map_err(|e| io::Error::new(io::ErrorKind::InvalidInput, format!("{}", e)))?
```

Comment:
maybe give a clear error message here

Reply:
After this PR is merged, I would commit anothor PR to make azblob support `encryption-scope`, and by the way I will make enhancement for azblob.

Reply:
Got, it will be updated in other prs.